### PR TITLE
Document the current architecture

### LINKED ISSUE #227: Document the current architecture
[Issuehunt badges](https://issuehunt.io/r/swapagarwal/swag-for-dev/issues/227)

A newbie needs to learn a bit of gulp, pug, styl. The doc can cover all things end-to-end from data.json to Netlify deployment.
This is important for onboarding new core contributors.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $20.00)

- [swapagarwal swapagarwal](https://issuehunt.io/u/swapagarwal) ($20.00)
### Submitted pull Requests
- [#398 Document the current architecture](https://issuehunt.io/r/swapagarwal/swag-for-dev/pull/398)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/swapagarwal/swag-for-dev/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by maybebored**
- [X] I've checked that this isn't a new swag opportunity proposal.
- [X] I've checked that this isn't a duplicate pull request.

Added `ARCHITECTURE.md` to document the current architecture. Closes #227
 

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#227: Document the current architecture](https://issuehunt.io/repos/121921584/issues/227)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Architecture of devSwag.io
+
+## Overview
+
+[devSwag.io](https://devswag.io/) uses [`gulp.js`](https://gulpjs.com/) to build the website. The website is composed of [`pug.js`](https://github.com/pugjs/pug) to template the front end of the website, [`Styl`](https://github.com/tj/styl) to add styling, and the backend is a simple JSON file called `data.json` that stores each swag item in a JSON format.
```
Comment (aslafy-z) [inline ARCHITECTURE.md]: - It's `Gulp`, not `gulp.js`
- It's `Pug`, not `pug.js`
- We use Stylus (https://github.com/stylus/stylus) to template style sheets.

File: ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+## Frontend
+
+[`pug.js`](https://github.com/pugjs/pug) is a template engine implemented using JavaScript for use in Node.js and the browser. More information can be found in their repository. `.pug` files are written using pug syntax which compiles to HTML.
```
Comment (aslafy-z) [inline ARCHITECTURE.md]: `Pug` again

File: ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+The devSwag frontend is composed of several `.pug` files stored in `src/pug`. These are later compiled to form static HTML files.
+
+## Backend
+
+Each event displayed at devSwag.io is a JSON object like this:
+
+```js
+{
+    "name": "7PHP", //Name of the event
```
Comment (aslafy-z) [inline ARCHITECTURE.md]:
```suggestion
    "name": "7PHP", // Name of the event
```

Comment (aslafy-z) [reply ARCHITECTURE.md]: Please use another example than PHP swag :pray: :rofl:

File: ARCHITECTURE.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+```
+Accompanying this is a script stored in `get-data.js` which extracts the image url of the event and generates unique filenames for each image file that will be downloaded later.
+
+## Build Process
+
+The entire website is built using gulp - an easy way to code your build instructions. These instructions are stored in `gulpfile.js`.
+
+In gulp each step is a task, represented as JavaScript function. [devSwag.io](https://devswag.io/)'s website is built by executing several gulp tasks sequentially. Dig into the [gulp documentation](https://gulpjs.com/docs/en/getting-started/quick-start) and `gulpfile.js` to get a better understanding of what is happening behind the hood. In summary however, two main tasks take place when `gulpfile.js` is invoked.
+
+    1. Downloading assets and compiling source code into a static website.
```
Comment (aslafy-z) [inline ARCHITECTURE.md]:
```suggestion
1. Downloading assets and compiling source code into a static website.
```

File: ARCHITECTURE.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    2. Starting a web server, to serve the compiled website.
```
Comment (aslafy-z) [inline ARCHITECTURE.md]:
```suggestion
2. Starting a web server, to serve the compiled website.
```

**Comment by maybebored** [comment]
@aslafy-z Resolved your comments.

**Comment by maybebored** [comment]
Bump!

**Comment by aslafy-z** [comment]
Thank you @maybebored. Please add a note in the readme referencing this `ARCHITECTURE.md` file and we'll be good to merge.

**Pull request merged**